### PR TITLE
Check if association already contains object 

### PR DESCRIPTION
**Pull request by klammbueddel**
This PR adds a test to reproduce #9579

### CODE REVIEW

**Comment by derrabus** [comment]
Please run `./vendor/bin/phpcbf` and commit the result. That should make the code style check happy.

**Comment by klammbueddel** [comment]
@derrabus: Thank you for the hint, I did that.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+/**
+ * @group GH-9579
+ */
+class GH9579Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]: Please use `$this->createSchemaForModels` instead, and remove `tearDown` (see https://github.com/doctrine/orm/pull/9526)

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        $part       = new GH9579Part();
+        $part->item = $item;
+
+        $this->_em->persist($item);
+        $this->_em->persist($container);
+        $this->_em->persist($part);
+        $this->_em->flush();
+        $this->_em->clear();
+    }
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php:50]: Why not move this inside `setUp()`?

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+    /**
+     * @group GH-9579
+     */
+    public function testIssue(): void
+    {
+        $this->createData();
+
+        $dql        = <<<DQL
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]:
```suggestion
        $dql        = <<<'DQL'
```

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    public function testIfAlreadyManaged(): void
+    {
+        $this->createData();
+
+        $containers = $this->_em->getRepository(GH9579Container::class)->findAll();
+        self::assertCount(1, $containers);
+        self::assertCount(1, $containers[0]->items);
+        self::assertCount(1, $containers[0]->items[0]->parts);
+
+        $dql        = <<<DQL
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]:
```suggestion
        $dql        = <<<'DQL'
```

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    /**
+     * @Id
+     * @Column(type="integer")
+     * @GeneratedValue
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var GH9579Item[]
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]:
```suggestion
     * @var Collection<int, GH9579Item>
```

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    /**
+     * @Id
+     * @Column(type="integer")
+     * @GeneratedValue
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var GH9579Part
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]:
```suggestion
     * @var Collection<int, GH9579Part>
```

**Comment by klammbueddel** [comment]
@greg0ire Thank you for the review. I updated the PR.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+class GH9579Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            GH9579Container::class,
+            GH9579Item::class,
+            GH9579Part::class,
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]: Trailing commas are not allowed on old versions of PHP
```suggestion
            GH9579Part::class
```

Comment (klammbueddel) [reply tests/Doctrine/Tests/ORM/Functional/Ticket/GH9579Test.php]: Yes, just saw this, too.. already fixed ;)

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @klammbueddel !

**Comment by derrabus** [comment]
Can we please change the PR title to something we want to read in the changelog? 🙂

**Pull request merged**